src/utils/common.js

```diff
@@ -589,3 +595,22 @@ export const formatTransactionFee = (feeInWei: string | number, gasToken: ?GasTo
 
   return `${formatAmount(utils.formatEther(feeInWei.toString()))} ETH`;
 };
+
+export const humanizeHexString = (hexString: ?string) => {
+  if (!hexString) return '';
+
+  const startCharsCount = 6;
+  const endCharsCount = 4;
+  const separator = '...';
+  const totalTruncatedSum = startCharsCount + endCharsCount + separator.length;
+
+  const words = hexString.toString().split(' ');
+  const firstWord = words[0];
+
+  if (words.length === 1) {
+    if (firstWord.length <= totalTruncatedSum) return firstWord;
+    return `${firstWord.slice(0, startCharsCount)}${separator}${firstWord.slice(-endCharsCount)}`;
+  }
+
+  return hexString;
```

Comment:
If `hexString` contains two or more words, none of it will be truncated. Is this expected behaviour?

Reply:
Hex string value cannot have space by design.